### PR TITLE
Gretel demo notebooks

### PR DESCRIPTION
**Pull request by mvansegbroeck**
Adding open in Colab button

### CODE REVIEW

**Review by johnnygreco — COMMENTED** [review]
Hey @mvansegbroeck, I added some comments in the first notebook. Not sure of the best way to review notebooks – experimented with adding them to the raw file. Looks like this wasn't a great idea, as the other notebook is packed into a single line 😫

Two general main comments:

- I think it would be helpful to have narratives that explain what's going on in a bit more detail. e.g., What's specifically is being demonstrated and why is it useful + different from other use cases (if multiple use cases shown). 

- The high-level interface is now pip installable with `gretel-client`. Do we want to update tabular notebooks to use it?

File: docs/notebooks/demo/gretel-demo-conditional-generation.ipynb [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from getpass import getpass\n",
+    "from gretel_client import configure_session\n",
+    "\n",
+    "gretel_endpoint = \"https://api.gretel.cloud\"\n",
+    "gretel_api_key = getpass(\"API Key: \")\n",
+    "\n",
+    "configure_session(\n",
```
[the remark this answers is not in the document]
Comment (mvansegbroeck) [reply docs/notebooks/demo/gretel-demo-conditional-generation.ipynb:51]: change to "prompt" and removed the other configure

File: docs/notebooks/demo/gretel-demo-conditional-generation.ipynb [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+    "    clear=True,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Fetch and prepare data\n",
+    "Read in the dataset as a Gretel Relational object"
```
[the remark this answers is not in the document]
Comment (mvansegbroeck) [reply docs/notebooks/demo/gretel-demo-conditional-generation.ipynb:64]: corrected

File: docs/notebooks/demo/gretel-demo-conditional-generation.ipynb [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+    "# Configure a Gretel session\n",
+    "\n",
+    "from gretel_client import configure_session\n",
+    "from gretel_client.projects import create_or_get_unique_project\n",
+    "\n",
+    "GRETEL_PROJECT_NAME = 'demo-conditional-generation'\n",
+    "\n",
+    "configure_session(\n",
+    "    api_key=\"prompt\",\n",
+    "    endpoint=\"https://api-dev.gretel.cloud\",\n",
```
[the remark this answers is not in the document]
Comment (mvansegbroeck) [reply docs/notebooks/demo/gretel-demo-conditional-generation.ipynb:104]: corrected

File: docs/notebooks/demo/gretel-demo-conditional-generation.ipynb [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "console_url = f\"https://console.gretel.ai/{project.project_guid}\"\n",
+    "print(console_url)"
```
[the remark this answers is not in the document]
Comment (mvansegbroeck) [reply docs/notebooks/demo/gretel-demo-conditional-generation.ipynb:169]: done

File: docs/notebooks/demo/gretel-demo-nested-json.ipynb
[the remark this answers is not in the document]
Comment (mvansegbroeck) [reply docs/notebooks/demo/gretel-demo-nested-json.ipynb:1]: Specific note: no I just change to "prompt".

[the remark this answers is not in the document]
Comment (mvansegbroeck) [reply docs/notebooks/demo/gretel-demo-nested-json.ipynb:1]: Added description to notebook on top.
I agree on the relational documentation. I am not sure if the current implementation for gretel relational on JSON is final (actually this is a bit of a workaround), so I suggest adding this later.

File: docs/notebooks/demo/gretel-demo-create-synthetic-dialogs.ipynb
[the remark this answers is not in the document]
Comment (mvansegbroeck) [reply docs/notebooks/demo/gretel-demo-create-synthetic-dialogs.ipynb:1]: Moved all helper functions into a single code block.

The blogpost will discuss the 2 datasets and I feel it is good to touch on the different types of datasets (daily conversation with emotions, question-answering, counseling). I am not sure if there is a better way to hide the conversions functions but w/o it the notebook is quite compact

**Review by johnnygreco — APPROVED** [review]
Looks good 👍

Not sure whether we should wait for the high-level interface in the tabular notebook. We'll probably update a bunch of notebooks once it's fully functional, so maybe we merge this and create a new PR to update later?

**Comment by mvansegbroeck** [comment]
> Looks good 👍
> 
> Not sure whether we should wait for the high-level interface in the tabular notebook. We'll probably update a bunch of notebooks once it's fully functional, so maybe we merge this and create a new PR to update later?

yep, that is the plan. Will do another PR for the SDK update.

File: docs/notebooks/demo/gretel-demo-conditional-generation.ipynb [hunk 5]
```diff
@@ -0,0 +1,426 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "col_value_counts = data_source[SEED_COLUMN].value_counts()\n",
+    "\n",
+    "seed_data = []\n",
```
[the remark this answers is not in the document]
Comment (mvansegbroeck) [reply docs/notebooks/demo/gretel-demo-conditional-generation.ipynb]: changed to this. Thanks

**Pull request merged**